R/mod_run_model_fix_params.R

```diff
@@ -40,6 +50,16 @@ mod_run_model_fix_params <- function(p, schema_text) {
   # these with NULL's as
   #   toJSON(NULL) == "{}"
   # nolint end
+
+  # Recursively nullify empty list elements
+  #
+  # Internal helper function to replace empty lists with NULL values for
+  # proper JSON serialization (empty objects instead of empty arrays).
+  #
+  # @param .x A list to process.
+  #
+  # @return The list with empty elements replaced by NULL.
+  # @noRd
```

Comment:
another internal function


```suggestion

  # Recursively nullify empty list elements
  #
  # Internal helper function to replace empty lists with NULL values for
  # proper JSON serialization (empty objects instead of empty arrays).
```